### ISSUE TITLE
Sidekicks Integration

### ISSUE THREAD

**Issue by DrmeddieWiese** [GregoryDosh/Home-Assistant-DreamScreen-Service#24]
Hi Gregory,

first thanks to your DreamScreen Service. Everything works great. 
Is there any solution to integrate Sidekicks? I know you haven't any hardware to test.
I have tried to integrate them in the configuration as own devices but it failed.

Thanks.

**Comment by J3n50m4t** [comment]
Any Logs?
I expect an unknown device id, which should be easy to das 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by DrmeddieWiese** [comment]
Thanks for interest :)

My log:
`Validation for Michsior14/ha-kaiterra failed.
21:17 custom_components/hacs/hacsbase/__init__.py (ERROR)

Repository is archived.
21:17 custom_components/hacs/repositories/integration.py (ERROR) - message first occured at 21:17 and shows up 2 times

Validation for Michsior14/ha-laser-egg failed.
21:17 custom_components/hacs/hacsbase/__init__.py (ERROR)

Setup of platform alexa_media is taking over 10 seconds.
21:17 __main__.py (WARNING) - message first occured at 21:17 and shows up 2 times

Setup of dreamscreen is taking over 10 seconds.
21:17 __main__.py (WARNING)

Failed to add device [192.168.178.49] dream_backleft. Try setting a 'timeout' in the device config.
21:17 custom_components/dreamscreen/__init__.py (WARNING)

couldn't get_state of 192.168.178.49
21:17 custom_components/dreamscreen/__init__.py (ERROR)

Failed to add device [192.168.178.47] dream_backright. Try setting a 'timeout' in the device config.
21:17 custom_components/dreamscreen/__init__.py (WARNING)

couldn't get_state of 192.168.178.47
21:17 custom_components/dreamscreen/__init__.py (ERROR)

Failed to add device [192.168.178.51] dream_frontleft. Try setting a 'timeout' in the device config.
21:16 custom_components/dreamscreen/__init__.py (WARNING)

couldn't get_state of 192.168.178.51
21:16 custom_components/dreamscreen/__init__.py (ERROR)

Failed to add device [192.168.178.48] dream_frontright. Try setting a 'timeout' in the device config.
21:16 custom_components/dreamscreen/__init__.py (WARNING)

couldn't get_state of 192.168.178.48
21:16 custom_components/dreamscreen/__init__.py (ERROR)`

**Comment by DrmeddieWiese** [comment]
Ok. I have now two sidekicks and the main screen as a group which can be turned on and off by voice (via USB Power plug). They are always in video mode. So they are only for movies.

The other two sidekicks are "stand-alone". Is there any option to control them with hassio? The best is to emulate them as hue light bulbs. Have anyone create such a solution?

**Comment by J3n50m4t** [comment]
So a group does work (kinda), but single sidekicks not? 
And then you also can't switch modes?

**Comment by DrmeddieWiese** [comment]
No in my case whether group or single sidekick can be discovered.

Thanks for reply

**Comment by J3n50m4t** [comment]
I assume same as @DrmeddieWiese  got, but with other names and ip? 

> Failed to add device [192.168.178.49] dream_backleft. Try setting a 'timeout' in the device config.
21:17 custom_components/dreamscreen/init.py (WARNING)

**Comment by DrmeddieWiese** [comment]
Yes. Same Issue. I've tried to change ip, name and factory reset but nothing happened. I added the port 8888 but this causes an error.

**Comment by J3n50m4t** [comment]
Could you add the [Logger Component](https://www.home-assistant.io/integrations/logger/) directly into your `configuration.yaml` and paste the logs? 

```
logger:
  default: critical
  logs:
    # log level for HA core just to get only erros and less output
    homeassistant.core: fatal
    # get all logs both components deliver
    homeassistant.components.dreamscreen: debug
    pydreamscreen: debug
```

**Comment by DrmeddieWiese** [comment]
Thanks for your help. I've add the logger component into my config. Here's my log.

`2019-11-24 20:04:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for dreamscreen which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-11-24 20:04:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for variable which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-11-24 20:04:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-11-24 20:04:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-11-24 20:04:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for var which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.

2019-11-24 20:04:20 DEBUG (MainThread) [pydreamscreen.devices] Listening...

2019-11-24 20:04:20 DEBUG (MainThread) [pydreamscreen.devices] Received message=b'\xfcC\xff`\x01\nSideKick Rechts\x00unassigned\x00\x00\x00\x00\x00\x00\x00\x00Q:\xffR\xe7\xee\xcb\x04\x0c\x01\x02\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x01\x00\x08\x03Z' from ip=192.168.178.73 port=[8888]

2019-11-24 20:04:23 DEBUG (MainThread) [pydreamscreen.devices] Received message=b'\xfc\x05\x010\x01\x0c\xff' from ip=192.168.178.46 port=[8888]

2019-11-24 20:04:28 ERROR (MainThread) [pydreamscreen.devices] couldn't get_state of 192.168.178.73'

**Comment by DrmeddieWiese** [comment]
Is there any possibilty to communicate in a direct way to a single sidekick? In relation to my log, the dreamscreen control is answering when I try to speak to a sidekick. Do I have to "speak" to the dreamscreen control to use a sidekick?
It's a little bit confusing because i haven't integrate the dreamscreen control in my config.yaml or grouped the sidekick which i try ro control.

**Comment by J3n50m4t** [comment]
I don't understand what you want to say. 
Is it possible to use a sidekick standalone with the Dreamscreen app? If yes we should be able to do this as well.

**Comment by DrmeddieWiese** [comment]
Yes it is possible to use a sidekick as standalone via App. The one which i try to control is a standalone and not grouped in the app.

Das andere schreibe ich jetzt mal auf Deutsch, da ich glaube, dass du mich verstehst ;)

Ich überlege ob es überhaupt möglich ist, einen Sidekick außerhalb der App direkt anzusprechen. Weil laut Log meine Anfrage an den Sidekick von der Haupt Dreamscreen Kontrolleinheit beantwortet wird. 
Mit der App funktioniert der alleinige Betrieb eines Sidekicks wunderbar. Ein Screenshot habe ich beigefügt
![Screenshot_20191124-221757_DreamScreen](https://user-images.githubusercontent.com/56825410/69501792-9b6d2d80-0f08-11ea-9c5a-589b26d118ba.jpg)

**Comment by J3n50m4t** [comment]
Ja ich verstehe das, allerdings sollten wir trotzdem weiter englisch nutzen, weil Gregory das hier sonst nicht versteht, und ggf hat der Die Lösung, der hat ja den Großteil hier gemacht.
According to their [api](http://dreamscreen.boards.net/thread/293/dreamscreen-wifi-udp-protocol) we should be able to read/write the states. 

>  Weil laut Log meine Anfrage an den Sidekick von der Haupt Dreamscreen Kontrolleinheit beantwortet wird.

Every Dreamscreen should respond to the discover command - and so does your sidekick right before the main Dreamscreen.

**Comment by DrmeddieWiese** [comment]
Ok. So the behaviour of my sidekick is right? It responds correctly. But i cannot understand the message. Even with the api

**Comment by eddyliao** [comment]
> 

Hey can you clarify more on this?
I have a DS + 2 sidekick in a single group
I am experiencing similar issues where only the main leds can be control but not the sidekicks

**Comment by ProLoser** [comment]
I sold my dreamscreen, but I flashed a custom firmware onto the sidekicks really easily and now they work great with home assistant out of the box

 https://www.google.com/url?sa=t&source=web&rct=j&url=https://m.youtube.com/watch%3Fv%3DfM10kmQmMIQ&ved=2ahUKEwiV3M6r-sf6AhUrlIkEHWuiDBMQjjh6BAgJEAE&usg=AOvVaw09UwI3FXa3jhsck69eM70r

**Comment by eddyliao** [comment]
Thanks i didnt know that exists, great find
